### PR TITLE
Create a repository for selecting player's matches using new db logic

### PR DESCRIPTION
**Pull request by shadre**

### CODE REVIEW

File: app/models/match.rb [hunk 1]
```diff
@@ -27,6 +27,10 @@ def teams
                .map { |_, group| group.map { |appearance| appearance.player } }
   end
 
+  def team(number)
+    number == 1 ? [player_1, player_2] : [player_3, player_4]
+  end
```
Comment (shadre) [inline app/models/match.rb]: ruby```
TEAM_1 = 1
TEAM_2 = 2

team == TEAM_1
```

File: app/services/player_matches_selector.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    Match.where(player_1: player)
+         .or(Match.where(player_2: player))
+         .or(Match.where(player_3: player))
+         .or(Match.where(player_4: player))
+         .where("created_at < ?", max_date)
+         .includes(:player_1, :player_2, :player_3, :player_4)
+         .first(matches_span)
+  end
+end
```
Comment (shadre) [inline app/services/player_matches_selector.rb]:
```ruby
class PlayerMatchesRepository
  def fetch_matches_for_player(player, exclude_last: 0)
  end
end
```

File: spec/services/player_matches_selector_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                         player_4: peter,
+                         first_team_score:  '2',
+                         second_team_score: '2') }
+
+  it 'selects all matches of a given player' do
+    annas_matches = described_class.new.call(anna)
+
+    expect(annas_matches.count).to eq(1)
+    expect(annas_matches).to include(match1)
+  end
```
Comment (shadre) [inline spec/services/player_matches_selector_spec.rb]:
```ruby
subject { described_class.new }

describe '#fetch_matches_for_player' do
  it '' do
    mark, joe, sherlock, anna, peter = create_list(:player, 5)
    match1 = create(:match, .....)

    subject.fetch_matches_for_player(anna)
  end
end
```

File: app/models/match.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 class Match < ApplicationRecord
+  TEAM_1 = 1
+  TEAM_2 = 2
```
Comment (AkkarinForest) [inline app/models/match.rb]: where is TEAM_2 used?

Comment (shadre) [reply app/models/match.rb]: It's not used anywhere currently.

Comment (marekciupak) [reply app/models/match.rb]: it could be used here https://github.com/LunarLogic/foosball-app/blob/master/app/services/match_creator.rb#L10

**Pull request merged**